### DISCUSSION TITLE
Add support for Azure Communications sms Provider

### DISCUSSION THREAD

**Discussion by giannik** [OrchardCMS/OrchardCore#15537] [Ideas]
@MikeAlhayek , Now that azure communications has been incorporated for email would it make sense to include an sms provider from Azure Communication Services. I've heard Twilio is rather expensive.

**Comment by hishamco** [comment]
Already implemented months ago,  a PR will be submitted in the upcoming days

**Comment by hishamco** [comment]
This is #15539 a draft PR, we might need a refactoring for the current `OC.SMS.Core` because it relies on Twilio. IMHO we could have

1. `OC.Sms.Abstractions`
2. `OC.Sms.Core`
3. `OC.Sms.Twilio`
4. `OC.Sms.Azure`

**Reply by MikeAlhayek** [reply]
No need for a new Twilio module. We do not add any special dependency for Twilio so it is fine to keep it in the Sms module. It's okay to create Sms.Azure module because you'll depend on a new Azure reference.

A good rule of thumb, create a module when you need to add external package as a new dependency.

**Reply by hishamco** [reply]
> No need for a new Twilio module

Why? there is no need to stick to the Twilio implementation by default

> A good rule of thumb, create a module when you need to add external package as a new dependency.

We already had a dependency when you created the `OC.Sms.Core` same thing as SMTP when it's created at the beginning
